packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -480,6 +577,51 @@ class FloatingActionButton extends StatelessWidget {
       ?? floatingActionButtonTheme.shape
       ?? (isExtended ? _defaultExtendedShape : _defaultShape);
 
+    BoxConstraints sizeConstraints;
+    Widget? resolvedChild = child;
+    switch(_floatingActionButtonType) {
+      case _FloatingActionButtonType.regular:
+        sizeConstraints = floatingActionButtonTheme.sizeConstraints ?? _sizeConstraints;
+        break;
+      case _FloatingActionButtonType.small:
+        sizeConstraints = floatingActionButtonTheme.smallSizeConstraints ?? _sizeConstraints;
+        break;
+      case _FloatingActionButtonType.large:
+        sizeConstraints = floatingActionButtonTheme.largeSizeConstraints ?? _sizeConstraints;
+        // The large FAB uses a larger icon.
+        resolvedChild = child != null ? IconTheme.merge(
+          data: const IconThemeData(size: 36.0),
+          child: child!,
+        ) : child;
+        break;
+      case _FloatingActionButtonType.extended:
+        sizeConstraints = floatingActionButtonTheme.extendedSizeConstraints ?? _sizeConstraints;
+        final double iconLabelSpacing = extendedIconLabelSpacing ?? floatingActionButtonTheme.extendedIconLabelSpacing ?? 8.0;
+        resolvedChild = _ChildOverflowBox(
+          child: Row(
```

Comment:
This is pretty hard to read. Here's an alternative:
```dart

const Widget width20 = SizedBox(width: 20.0);
const Widget width16 = SizedBox(width: 16.0);

  child: Row(
    mainAxisSize: MainAxisSize.min,
    children: child == null
        ? <Widget>[width20, _label!, width20]
        : isExtended
           ? <Widget>[width16, child!, SizedBox(width: iconLabelSpacing), _label!, width20]
           : <Widget>[width20, child!, width20];
  )
```

Reply:
Done